### DISCUSSION TITLE
Allow selecting a schema in database connection string settings

### DISCUSSION THREAD

**Discussion by smonolo** [supabase/supabase#22192] [Feature Requests]
Database connection strings accept a `schema` parameter that allows specifying the schema to use. In the database settings page, under connection string info, it's currently not poissible to select a schema and get the automatically generated string that includes the parameter if needed.

As a user, I want to be able to choose a specific database schema and get the correct connection string that includes the `schema` parameter if the selection differs from `public`.

Here's a screenshot of what I'm referring to.

![image](https://github.com/supabase/supabase/assets/29622224/fdaae72b-8411-4a8f-bac6-ab7c37c0dbd1)

Let me know if you have any questions.

**Comment by piotrek-toolla** [comment]
+1 I would say that postgresql connection URI typically supports param specs in `name=value` form. This makes it possible to set the schema as
`postgres://user:pass@host:port/db?options=--search_path%3Dmyschema`.
But it doesn't work with Supabase. Can you please support options in connection URI? Alternatively, please share a workaround.

**Reply by b3nab** [reply]
Yes I'm having the same "issue" trying to find a way to set search_schema using the connection string. Did you find a solution?

**Reply by smonolo** [reply]
@b3nab I believe Supabase already supports specifying a schema by attaching the query parameter to the final URL like this: `postgres://{...}?schema={schema}` (e.g. `?schema=staging`).

If I remember correctly, in this discussion I was referring mainly to the dashboard part, so the UI, as in some cases a user 1. might not know they can specify a schema and 2. they might just want to get the string quickly.

What do you think?

CC: @piotrek-toolla

**Reply by bradleyhodges** [reply]
> @b3nab I believe Supabase already supports specifying a schema by attaching the query parameter to the final URL like this: `postgres://{...}?schema={schema}` (e.g. `?schema=staging`).

It seems that, at least as of February 2025, this is not the case. Neither `?schema` nor `currentSchema` appear to work when connecting to a Supabase database:

```
psql: error: invalid URI query parameter: "schema"
psql: error: invalid URI query parameter: "currentSchema"
```

**Reply by smonolo** [reply]
@bradleyhodges That's weird, I had some projects that used to work with the solution I suggested, or something similar to that. What's the chance that they renamed or moved the config? I'm not really up-to-date with Supabase changes, so maybe I missed the changelog.

I'll go check the env files of that project later and come back here with an update. But looking at the documentation might be the way to go at this point.

**Reply by smonolo** [reply]
@bradleyhodges Here you go: `postgres://[...]/postgres?schema=staging`

This worked fine about a year ago using Prisma on an Express application, not sure about now.
